php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+        f.blockNamespaceStatement(
+          b.token(PHPKeyword.NAMESPACE),
+          b.optional(NAMESPACE_NAME()),
+          b.token(LCURLYBRACE),
+          b.zeroOrMore(TOP_STATEMENT()),
+          b.token(RCURLYBRACE))));
+  }
+
+  public UseStatementTree USE_STATEMENT() {
+    return b.<UseStatementTree>nonterminal(PHPLexicalGrammar.USE_STATEMENT).is(
+      f.useStatement(
+        b.token(PHPKeyword.USE),
+        b.optional(
+          b.firstOf(
+            b.token(PHPKeyword.CONST),
+            b.token(PHPKeyword.FUNCTION))),
+        USE_CLAUSE(),
+        b.zeroOrMore(f.newTuple90(b.token(PHPPunctuator.COMMA), USE_CLAUSE())),
+        EOS()));
+  }
+
+  public StatementTree STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.STATEMENT).is(
+      b.firstOf(
+        BLOCK(),
+        THROW_STATEMENT(),
+        IF_STATEMENT(),
+        WHILE_STATEMENT(),
+        DO_WHILE_STATEMENT(),
+        FOREACH_STATEMENT(),
+        FOR_STATEMENT(),
+        SWITCH_STATEMENT(),
+        BREAK_STATEMENT(),
+        CONTINUE_STATEMENT(),
+        RETURN_STATEMENT(),
+        EMPTY_STATEMENT(),
+        YIELD_STATEMENT(),
+        GLOBAL_STATEMENT(),
+        STATIC_STATEMENT(),
+        ECHO_STATEMENT(),
+        TRY_STATEMENT(),
+        DECLARE_STATEMENT(),
+        GOTO_STATEMENT(),
+        INLINE_HTML(),
+        UNSET_VARIABLE_STATEMENT(),
+        EXPRESSION_STATEMENT(),
+        LABEL()));
+  }
+
+  public ExpressionStatementTree ECHO_STATEMENT() {
+    return b.<ExpressionStatementTree>nonterminal().is(
+      f.echoStatement(
+        b.token(ECHO),
+        EXPRESSION(),
+        b.zeroOrMore(f.newTuple54(b.token(COMMA), EXPRESSION())),
+        EOS()
+      )
+    );
+  }
+
+  public StaticStatementTree STATIC_STATEMENT() {
+    return b.<StaticStatementTree>nonterminal(PHPLexicalGrammar.STATIC_STATEMENT).is(
+      f.staticStatement(
+        b.token(STATIC),
+        STATIC_VAR(),
+        b.zeroOrMore(f.newTuple22(b.token(COMMA), STATIC_VAR())),
+        EOS()));
+  }
+
+  public VariableDeclarationTree STATIC_VAR() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.STATIC_VAR).is(
+      f.staticVar(
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(f.newTuple24(b.token(EQU), STATIC_SCALAR()))));
+  }
+
+  public DeclareStatementTree DECLARE_STATEMENT() {
+    return b.<DeclareStatementTree>nonterminal(PHPLexicalGrammar.DECLARE_STATEMENT).is(
+      b.firstOf(
+        f.shortDeclareStatement(
+          DECLARE_STATEMENT_HEAD(),
+          EOS()
+        ),
+        f.declareStatementWithOneStatement(
+          DECLARE_STATEMENT_HEAD(),
+          STATEMENT()
+        ),
+        f.alternativeDeclareStatement(
+          DECLARE_STATEMENT_HEAD(),
+          b.token(COLON),
+          b.zeroOrMore(INNER_STATEMENT()),
+          b.token(PHPKeyword.ENDDECLARE),
+          EOS())));
+  }
+
+  public DeclareStatementHead DECLARE_STATEMENT_HEAD() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)